pkg/domain/resourcegroup/runaway.go

```diff
@@ -499,50 +483,48 @@ func (r *RunawayChecker) BeforeExecutor() error {
 	if r == nil {
 		return nil
 	}
+	// Check if the query matches any specified watch rules.
 	for _, convict := range r.getConvictIdentifiers() {
 		watched, action := r.manager.examineWatchList(r.resourceGroupName, convict)
-		if watched {
-			if action == rmpb.RunawayAction_NoneAction && r.setting != nil {
-				action = r.setting.Action
-			}
-			r.markedByWatch = true
-			now := time.Now()
-			r.watchAction = action
-			r.markRunaway(RunawayMatchTypeWatch, action, &now)
-			// If no match action, it will do nothing.
-			switch action {
-			case rmpb.RunawayAction_Kill:
-				return exeerrors.ErrResourceGroupQueryRunawayQuarantine
-			case rmpb.RunawayAction_CoolDown:
-				// This action should be done in BeforeCopRequest.
-				return nil
-			case rmpb.RunawayAction_DryRun:
-				return nil
```

Comment:
why removed `RunawayAction_DryRun`?

Reply:
It can be covered by the `default` case, there is no need to explicitly write it here I think

Reply:
I agree that this action can be similar to the default case.
> Previous behavior may have been wrong:
> Previously, the dryRun branch would return directly, but now it iterates over all identifiers in a for loop.

There is a concept of [dryRun](https://docs.pingcap.com/tidb/stable/tidb-resource-control#query_limit-parameters) in the documentation, but after removing this `RunawayAction_DryRun` in the code,  there is no dryRun to be found in the code (which may be confusing?). Do we need to put it here explicitly? Just like: 
```
		case rmpb.RunawayAction_DryRun:
		default:
```

Reply:
Good catch, you're right!